ui/ui.js

```diff
@@ -485,17 +485,21 @@ shaka.ui.Overlay = class {
     //    <source src='foo.m2u8'/>
     //  </video>
     // It should not be specified on both.
+    const urls = [];
     const src = video.getAttribute('src');
     if (src) {
-      const sourceElem = document.createElement('source');
-      sourceElem.setAttribute('src', src);
-      video.appendChild(sourceElem);
+      urls.push(src);
       video.removeAttribute('src');
     }
+    for (const source of /** @type {!NodeList<!HTMLSourceElement>}*/(
```

Comment:
Could you store the value of this NodeList, and then use the stored NodeList in the for loop? `const sources` or something. The formatting as-is puts the `video.getElementsByTagName` call on the same indentation level as the contents of the loop, which is hard to read.